patch/kconfig-secure-boot-exclusions

```diff
@@ -0,0 +1,21 @@
+[common]
+
+[amd64]
+CONFIG_MODULE_SIG_SHA256
+# For mellanox
```

Comment:
why these are excluded? can we have more explanation for the justifications?

Reply:
SHA256 is excluded because we are using SHA512. Its more secure.
there is more description in the HLD link attached in the description of this PR.
About the lockdown, we have a plan to support it in the feature, for now for Mellanox its disabled.

Reply:
> I'm planning to move the SHA512 config to apply to all kernel builds in a future PR; this isn't necessarily secure-boot specific, and is nice to have in general.

Sound good, can we for now save the PR as is, and when you modify the general config you can remove it.

Because we are implicitly signing the kernel modules with sha512 in the sonic-buildimage:
https://github.com/sonic-net/sonic-buildimage/pull/12692/files#diff-de80d4961ffb88d808888c6d160af8717e70ec6c21675b0f5124b0d27db7a166 
So, if the kernel configuration does not match, the image will not boot.